src/connector/src/connector_common/iceberg/jni_catalog.rs

```diff
@@ -402,8 +312,57 @@ impl CatalogV2 for JniCatalog {
     }
 
     /// Update a table to the catalog.
-    async fn update_table(&self, _commit: TableCommit) -> iceberg::Result<TableV2> {
-        todo!()
+    async fn update_table(&self, mut commit: TableCommit) -> iceberg::Result<Table> {
+        execute_with_jni_env(self.jvm, |env| {
+            let requirements = commit.take_requirements();
+            let updates = commit.take_updates();
+            let request = CommitTableRequest {
+                identifier: commit.identifier().clone(),
+                requirements,
+                updates,
+            };
+            let request_str = serde_json::to_string(&request)?;
+
+            let request_jni_str = env.new_string(&request_str).with_context(|| {
+                format!("Failed to create jni string from request json: {request_str}.")
+            })?;
+
+            let result_json =
+                call_method!(env, self.java_catalog.as_obj(), {String updateTable(String)},
+                &request_jni_str)
+                .with_context(|| {
+                    format!("Failed to update iceberg table: {}", commit.identifier())
+                })?;
+
+            let rust_json_str = jobj_to_str(env, result_json)?;
+
+            let response: CommitTableResponse = serde_json::from_str(&rust_json_str)?;
+
+            tracing::info!(
+                "Table metadata location of {} is {}",
+                commit.identifier(),
+                response.metadata_location
+            );
+
+            let table_metadata = response.metadata;
+
+            let file_io = FileIO::from_path(&response.metadata_location)?
+                .with_props(self.config.table_io_configs.iter())
```

Comment:
It seems we don't need `self.config` (BaseCatalogConfig is a struct from icelake) anymore, we can simplify it to a map.